File: leet_code_algorithms/add_binary/add_binary.ts
const addBinary = (a: string, b: string): string => {
  let total = "";
  let carry = 0;
  const range = a.length > b.length ? a.length : b.length;
  for (let i = 0; i < range; i++) {
    const int_a = i < a.length ? parseInt(a[a.length - 1 - i]) : 0;
    const int_b = i < b.length ? parseInt(b[b.length - 1 - i]) : 0;
    const sum = int_a + int_b + carry;
    total = (sum % 2).toString() + total;
    carry = Math.floor(sum / 2);
  }
  if (carry) {
    total = "1" + total;
  }
  return total;
};

File: leet_code_algorithms/add_binary/Add_binary.java
package add_binary;

public class Add_binary {
    public String addBinary(String a, String b) {
        String total = "";
        int carry = 0;
        int range;
        if (a.length() >= b.length()) {
            range = a.length();
        } else {
            range = b.length();
        }
        for (var i = 0; i < range; i++) {
            int int_a;
            int int_b;
            if (i < a.length()) {
                int_a = Integer.parseInt(String.valueOf(a.charAt(a.length() - 1 - i)));
            } else {
                int_a = 0;
            }
            if (i < b.length()) {
                int_b = Integer.parseInt(String.valueOf(b.charAt(b.length() - 1 - i)));
            } else {
                int_b = 0;
            }
            int sum = int_a + int_b + carry;
            total = String.valueOf(sum % 2) + total;
            carry = Math.floorDiv(sum, 2);
        }
        if (carry == 1) {
            total = "1" + total;
        }
        return total;
    }
}
